extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php

```diff
@@ -522,8 +526,9 @@ private function saveUsersAvatar( $userId = null, $data = null ) {
 			if ( empty( $wgAvatarsUseService ) ) {
 				// TODO: $user->getTouched() get be used to invalidate avatar URLs instead
 				$user->setGlobalAttribute( 'avatar_rev', date( 'U' ) );
-				$user->saveSettings();
 			}
+
+			$user->saveSettings();
```

Comment:
Fixed in 4b9b338 (added the call to `User::saveSettings` when setting the full URL of the predefined avata in a23ad91)